File: scratch/clean_localstorage.js
const fs = require('fs');
const path = require('path');

const files = [
    path.join(__dirname, '../frontend/src/pages/VendorDashboard.jsx'),
    path.join(__dirname, '../frontend/src/pages/UserDashboard.jsx'),
    path.join(__dirname, '../frontend/src/pages/AdminDashboard.jsx'),
    path.join(__dirname, '../frontend/src/pages/Marketplace.jsx'),
    path.join(__dirname, '../frontend/src/components/DashboardLayout.jsx'),
    path.join(__dirname, '../frontend/src/components/UserSidebar.jsx')
];

function replaceInFile(filepath) {
    if (!fs.existsSync(filepath)) return;
    
    let content = fs.readFileSync(filepath, 'utf8');

    // Replace JSON.parse(localStorage.getItem('mock...')) || '[]' with []
    content = content.replace(/JSON\.parse\(localStorage\.getItem\('mock[^']+'\) \|\| '\[\]'\)/g, "[]");
    
    // Replace localStorage.getItem('mock...') with null
    content = content.replace(/localStorage\.getItem\('mock[^']+'\)/g, "null");

    // Replace localStorage.setItem('mock...', <anything>); across multiple lines
    // We use a regex that matches localStorage.setItem('mock...', up to the closing );
    // Be careful with JSON.stringify() which might contain ); inside it.
    // A better approach is matching from localStorage.setItem('mock up to the next semicolon.
    // However, JS might not have semicolons. Let's just match single line ones for now.
    // Or we can use regex: /localStorage\.setItem\('mock[^']+',[\s\S]*?\);/g
    // This could be greedy and match too much if there are multiple statements!
    // A safer regex for multiline setItem:
    content = content.replace(/localStorage\.setItem\('mock[^']+',[^;]+;/g, "");
    
    // Some lines might not end with a semicolon.
    content = content.replace(/localStorage\.setItem\('mock[a-zA-Z0-9]+',\s*JSON\.stringify\([^)]+\)\)(;?)/g, "");

    // Let's remove mock_vendor_id_123 fallback
    content = content.replace(/ \|\| 'mock_vendor_id_123'/g, "");
    content = content.replace(/ \? '_id' : 'mock_vendor_id_123'/g, "");

    fs.writeFileSync(filepath, content, 'utf8');
    console.log(`Processed ${filepath}`);
}

files.forEach(replaceInFile);
console.log("Done.");

File: scratch/clean_localstorage.py
import re
import glob
import os

files = [
    r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\frontend\src\pages\VendorDashboard.jsx",
    r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\frontend\src\pages\UserDashboard.jsx",
    r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\frontend\src\pages\AdminDashboard.jsx",
    r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\frontend\src\pages\Marketplace.jsx",
    r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\frontend\src\components\DashboardLayout.jsx",
    r"c:\Users\renug\OneDrive\Desktop\AI Interior Final Project\frontend\src\components\UserSidebar.jsx"
]

def replace_in_file(filepath):
    if not os.path.exists(filepath):
        return
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Replace localStorage.getItem for mock data with empty array
    # e.g., JSON.parse(localStorage.getItem('mockOrders') || '[]') -> []
    content = re.sub(r"JSON\.parse\(localStorage\.getItem\('mock[^']+'\) \|\| '\[\]'\)", "[]", content)
    content = re.sub(r"localStorage\.getItem\('mock[^']+'\)", "null", content)
    
    # We want to remove localStorage.setItem('mock...', ...) lines
    # It's better to replace them with a console.log or just an empty statement to avoid syntax errors if they are in an expression.
    content = re.sub(r"localStorage\.setItem\('mock[^']+',[^;]+;", "", content)

    # Some setItems span multiple lines. For those, we can just look for them.
    # A simple regex to remove localStorage.setItem('mock...', <anything>);
    content = re.sub(r"localStorage\.setItem\('mock[^']+',[\s\S]*?\);", "", content)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(content)
        
for f in files:
    replace_in_file(f)

print("Replacement complete.")
